tests/bsim/bluetooth/mesh/src/mesh_test.c

```diff
@@ -546,6 +546,54 @@ uint16_t bt_mesh_test_own_addr_get(uint16_t start_addr)
 	return start_addr + get_device_nbr();
 }
 
+void bt_mesh_test_send_over_adv(void *data, size_t len)
+{
+	struct bt_mesh_adv *adv = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+						     BT_MESH_TRANSMIT(0, 20), K_NO_WAIT);
+	net_buf_simple_add_mem(&adv->b, data, len);
+	bt_mesh_adv_send(adv, NULL, NULL);
+}
+
+int bt_mesh_test_wait_for_adv(bt_le_scan_cb_t scan_cb, struct k_sem *observer_sem, uint16_t wait)
+{
+	struct bt_le_scan_param scan_param = {
+		.type       = BT_HCI_LE_SCAN_PASSIVE,
+		.options    = BT_LE_SCAN_OPT_NONE,
+		.interval   = BT_MESH_ADV_SCAN_UNIT(1000),
+		.window     = BT_MESH_ADV_SCAN_UNIT(1000)
+	};
+	int err;
+
+	err = bt_le_scan_start(&scan_param, scan_cb);
+	if (err && err != -EALREADY) {
+		LOG_ERR("Starting scan failed (err %d)", err);
+		return err;
+	}
+
+	err = k_sem_take(observer_sem, K_SECONDS(wait));
+	if (err == -EAGAIN) {
+		LOG_WRN("Taking sem timed out (err %d)", err);
+		err = bt_le_scan_stop();
```

Comment:
Hm... could you just create one more local variable like `returned_value` and assign here to `-EAGAIN`? Or something similar.
Seems like unnecessary duplication for scanner stopping.